collector/collector.cpp

```diff
@@ -107,9 +107,15 @@ int InsertModule(int fd, const std::unordered_map<std::string, std::string>& arg
     args_str += entry.first + "=" + entry.second;
   }
   CLOG(DEBUG) << "Kernel module arguments: " << args_str;
-  int res = finit_module(fd, args_str.c_str(), 0);
-  if (res != 0) return res;
   struct stat st;
+  fstat(fd, &st);
+  size_t image_size = st.st_size;
+  void* image = malloc(image_size);
+  lseek(fd, 0, SEEK_SET);
+  read(fd, image, image_size);
```

Comment:
And while we are at it, It'd also be nice to check if the `read` bytes are equal to the expected size and no error occurred here.

Reply:
We could also check the return code for `lseek` but if that failed, `read` will probably fail itself or read less bytes than expected.